### PR TITLE
community: Improve Kubeflow Ecosystem diagram

### PR DESCRIPTION
**Pull request by thesuperzapper**
/area website

This PR splits out the "Kubeflow Ecosystem" diagram updates from https://github.com/kubeflow/website/pull/3981 so we can seperate the discussions about each.

The reasons for the change:

- make the diagram more "square" (so its more easy to use in a slide deck)
- make padding more consistent
- make colors more distinct
- remove "BentoML" from "External Add-Ons" (as we already did on the website)
- __NEW:__ I have removed the "istio/cert-manager" sections and simply placed a horizontal "kubernetes" layer above the "platforms" (aws, gcp, on-prem).

### See: [Files Changed](https://github.com/kubeflow/website/pull/4038/files)

### CODE REVIEW

**Comment by thesuperzapper** [comment]
@andreyvelich @franciscojavierarceo @varodrig these updates significantly improve the architeture diagram and make it easier to use in presentations.

It does not significantly change the content (other than removing BentoML), so as long as we agree on the new "section names" (left side), I think we can merge.

**Comment by franciscojavierarceo** [comment]
Yeah, it looks good to me. Will defer to @andreyvelich

**Comment by varodrig** [comment]
/lgtm 

thank you for working on this.

**Comment by varodrig** [comment]
@juliusvonkohout 
cc @thesuperzapper

**Review by andreyvelich — COMMENTED** [review]
Thank you @thesuperzapper!
I will take a look soon.
/hold for review

**Comment by juliusvonkohout** [comment]
@thesuperzapper why did you change the description on the left. This looks like a significant change to me. Is it possible to stay closer to the original? The terms Hardware and infrastructureseem better to me than chips and platform.

**Comment by thesuperzapper** [comment]
> @thesuperzapper why did you change the description on the left. This looks like a significant change to me. Is it possible to stay closer to the original? The terms Hardware and infrastructureseem better to me than chips and platform.

It's simply that "Hardware" is too long to fit vertically, but I think "Chips" also makes a lot of sense for the bottom layer.

But I think that "Platforms" is a more correct term to refefer to AWS/GCP/On-Prem, but we can discuss this further.

File: content/en/docs/started/images/kubeflow-architecture.drawio.svg
Comment (andreyvelich) [inline content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: A few things:
1. Should we combine circles for Kubernetes and Cloud icons ? I don't feel a strong need to separate them in 2 separate boxes under Platforms.
2. We discussed many times with @kubeflow/kubeflow-steering-committee that we want to remove External Add-Ons sections and replace it with the Kubeflow website page called: `Integrations with Kubeflow Projects`.
In this page, all projects that have integrations with Kubeflow Ecosystem can add themself.
3. Can we rename Kubeflow Components to Kubeflow Projects ?
4. What do you think about changing the Frameworks to AI/ML Ecosystem ? I understand that not all of them only related to  AI/ML, but the primary goal of Kubeflow is to stitch ML and Cloud Ecosystems together.

**Review by andreyvelich — COMMENTED** [review]
Thank you for this @thesuperzapper!

File: content/en/docs/started/images/kubeflow-architecture.drawio.svg
Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: >I like the seperation because it shows that Kubernetes sits "on top" of all the platforms.

Do you prefer it over the blue line that we have before ?

>Lets only remove the external add-ons from the diagram once we remove them from the website, otherwise we will be dropping feast from the diagram.

Alright, we can do this as a followup.

>We have always used the term "Components", and this would need to be a wider discussion:

Sure, we can do this as a followup, but for me calling Kubeflow "Ecosystem" and tools we develop under "Ecosystem" Kubeflow Project makes sense.
It also aligns with PyTorch Ecosystem: https://pytorch.org/ecosystem/ 

>Alternatively, we could just extend the "tools" section to cover all them, because its not really correct to call Jupyter a framework.

In that case, how we can distinguish Kubeflow Projects with ML Ecosystem ?

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I would call Jupyter framework since it is much more than UI and IDE. It also provides set of protocols to talk to Kernels: https://docs.jupyter.org/en/latest/what_is_jupyter.html#talking-to-kernels-the-jupyter-protocol

cc @zsailer

Comment (vikas-saxena02) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I think its worth keeping istio and dex as they are key components of the Kubeflow Ecosystem

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I would disagree with it, Istio and Dex are optional components, and not required for all users to use Kubeflow Ecosystem. For example, we know users who prefer Cilium over Istio and their custom identity providers.
So from my point of view, we should only keep Kubernetes on top of Cloud Providers.

Comment (juliusvonkohout) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: In probably 95+% of full platform installations it is istio and or Istio with just cilium as CNI plugin. The same goes for oauth2-proxy and Dex. But yes, it is not that end user relevant to mention it in the diagram.

Comment (andreyvelich) [inline content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: @thesuperzapper Shell we rename ?
```
Tools --> Kubeflow Projects 
Frameworks --> AI/ML Ecosystem
```

That will also align with changes we make here, to rename Kubeflow Components to Kubeflow Projects:
https://github.com/kubeflow/website/pull/4023

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: cc @franciscojavierarceo

**Review by andreyvelich — COMMENTED** [review]
cc WGs for the feedback on these changes.
@kubeflow/wg-training-leads @kubeflow/wg-data-leads @kubeflow/wg-pipeline-leads @kubeflow/wg-automl-leads @kubeflow/wg-notebooks-leads @kubeflow/wg-manifests-leads @kubeflow/kubeflow-steering-committee

File: content/en/docs/started/images/kubeflow-architecture.drawio.svg
Comment (franciscojavierarceo) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I am pro renaming to "Kubeflow Projects" and I like AI/ML Ecosystem.

I don't love the gray color of the "Chips, Platforms, Tools, Frameworks" though. 😅

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: @thesuperzapper Can you address this suggestion, so we can move forward with your new diagram ?

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: For the gray colors, can we try to use Kubeflow color and see how it looks ?
https://github.com/kubeflow/marketing-materials/blob/master/logos/Logo-File-Guide.pdf

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: @andreyvelich I have made the sidebars light grey, as the blue colors were too vibrant.

I have also renamed the heading to "Kubeflow Projects".

The side headings are now:

- Intergrations (for libraries/IDEs - same as before this PR)
- Components (to cover both projects and add-ons that are "part of kubeflow")
- Infrastructure (to cover everything from chips to Kubernetes)

I have also added an underline to the in-box headings.

I have also added "others..." to the "integrations" to show that the list is not comprehensive.

**Comment by thesuperzapper** [comment]
@andreyvelich @franciscojavierarceo I think this is ready for another review.

I would like to get this one updated so @varodrig can use the updated one in the Kubecon slides.

**Review by terrytangyuan — COMMENTED** [review]
What's the rationale behind whether to have a "kubeflow" prefix for each project?

**Comment by thesuperzapper** [comment]
> What's the rationale behind whether to have a "kubeflow" prefix for each project?

@terrytangyuan Because these are the actual "names" of each project, with exception of "Katib" and "KServe" which don't stylize with the "Kubeflow ..." prefix.

Many of our project names would be very generic without the stylization, e.g. "Notebooks" or "Pipelines", or "Model Registry".

File: content/en/docs/started/images/kubeflow-architecture.drawio.svg
Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: >Components (to cover both projects and add-ons that are "part of kubeflow")

Since we are going to remove external add-ons from the Kubeflow Ecosystem diagram and Kubeflow website, what should be instead of Components ?


Also, why you don't want to rename Integration to AI/ML Ecosystem and @franciscojavierarceo and I proposed above.

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I really want to keep our links to KServe, Elyra, and Feast on the website.

Are you sure you want to remove them from the website?

I really think its important to have non-kubeflow tools have some presence on the website (especially those ones).

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: After discussing this with the community, we’ve clarified the purpose of the Ecosystem diagram: to help users understand what's inside the Kubeflow Ecosystem. So they have clear understanding: "What is Kubeflow?"

We still want to highlight third-party integrations with Kubeflow Projects, but these should have a dedicated page on the website rather than being included in the Ecosystem diagram.

One idea that comes to mind is creating an **Awesome Kubeflow Integrations** page on the Kubeflow website. 
So folks can just update this page to include their integrations.
@terrytangyuan already have this GitHub repo that we can take some projects: https://github.com/terrytangyuan/awesome-kubeflow

cc @kubeflow/kubeflow-steering-committee @jbottum @varodrig @rimolive

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: Yes, I suggest to remove it from the ecosystem diagram, so we make it very clear for users:
"What are Kubeflow Projects" ?
Those 3rd party integration can leave in a separate page.

Comment (franciscojavierarceo) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I would prefer to hold off on removing Feast since it's attempting to be donated. :)

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I think in the interest of improving the diagram for Kubecon slides, let's leave the external-addons until we have had a larger discussion about what to do with them.

If people agree that the currently proposed diagram updates are good, lets merge.

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I am fine to discuss this in a followup.
@thesuperzapper Can you at least rename Integrations to AI/ML Ecosystem in this diagram as we suggested with @franciscojavierarceo ?

**Comment by tarilabs** [comment]
fwiw I believe the prefix can be omitted if there is a wat to distinguish KServe add-on, which is currently in the same box as other KF projects, my2c

**Comment by thesuperzapper** [comment]
> fwiw I believe the prefix can be omitted if there is a wat to distinguish KServe add-on, which is currently in the same box as other KF projects, my2c

I strongly disagree, it's critical that we consistently brand our tools. For example, the name of our pipeline tool is "Kubeflow Pipelines" not "Pipelines".

File: content/en/docs/started/images/kubeflow-architecture.drawio.svg
Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: It is important to say that Kubeflow is a bridge between AIML and Cloud ecosystems.
If you don't like `/` let's just use `AI Ecosystem`. WDYT @franciscojavierarceo ?

**Comment by tarilabs** [comment]
> > fwiw I believe the prefix can be omitted if there is a wat to distinguish KServe add-on, which is currently in the same box as other KF projects, my2c
> 
> I strongly disagree, it's critical that we consistently brand our tools. For example, the name of our pipeline tool is "Kubeflow Pipelines" not "Pipelines".

That's fine :) the actual point in my comment however, I thought it was important that someone might confuse KServe for being "inside" Kubeflow (the "External Add-on" is not retained from the old diagram) for a Kubeflow hosted project (based label of the frame which encloses it).

If nobody feel this might be confusing for new users, ignore my comment altogether! 👍

**Comment by thesuperzapper** [comment]
> >I strongly disagree, it's critical that we consistently brand our tools. For example, the name of our pipeline tool is "Kubeflow Pipelines" not "Pipelines".
> 
> We can name everything with Kubeflow prefix. I believe, we should be good to use Kubeflow Katib and Kubeflow KServe in this diagram.

@andreyvelich Every time we have renamed Katib to "Kubeflow Katib", people end up reverting it, as far as I can tell, the official name is "Katib".

It is also less of a problem than pipelines/notebooks/etc because Katib is quite a distinct name.

**Comment by andreyvelich** [comment]
>@andreyvelich Every time we have renamed Katib to "Kubeflow Katib", people end up reverting it, as far as I can tell, the official name is "Katib".

It is also less of a problem than pipelines/notebooks/etc because Katib is quite a distinct name.

On behalf of Katib project, I can say that it is ok if we are going to use Kubeflow Katib name. cc @kubeflow/wg-automl-leads 
@Electronic-Waste @helenxie-bit.
Additionally, in the near future we are going to re-brand Katib project to Kubeflow Optimizer to be aligned with Kubeflow SDK naming: https://github.com/kubeflow/community/pull/823.
So it is totally fine if we keep Kubeflow Katib in the ecosystem diagram for now.

File: content/en/docs/started/images/kubeflow-architecture.drawio.svg
Comment (franciscojavierarceo) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: Lgtm

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-architecture.drawio.svg:1]: I have updated it to say "AI Ecosystem" with two spaces so its easier to read.

**Review by andreyvelich — COMMENTED** [review]
Thank you @thesuperzapper!
I think, we should be good to move this forward.
/lgtm
/hold for others to review
/assign @franciscojavierarceo @juliusvonkohout @johnugeorge @terrytangyuan @tarilabs

**Review by andreyvelich — COMMENTED** [review]
Thanks for this effort @thesuperzapper!
/hold cancel
